controller/api/public/grpc_server.go

```diff
@@ -128,6 +132,32 @@ func (s *grpcServer) ListPods(ctx context.Context, req *pb.Empty) (*pb.ListPodsR
 	return resp, nil
 }
 
+func (s *grpcServer) SelfCheck(ctx context.Context, in *common.SelfCheckRequest) (*common.SelfCheckResponse, error) {
+	telemetryClientCheck := &common.CheckResult{
+		SubsystemName:    TelemetryClientSubsystemName,
+		CheckDescription: TelemetryClientCheckDescription,
+		Status:           string(healthcheck.CheckError),
+	}
+
+	_, err := s.telemetryClient.ListPods(ctx, &telemPb.ListPodsRequest{})
+	if err != nil {
+		telemetryClientCheck.Status = string(healthcheck.CheckError)
+		telemetryClientCheck.FriendlyMessageToUser = fmt.Sprintf("Error talking to telemetry service from control plane: %s", err.Error())
+		return nil, err
```

Comment:
if `s.telemetryClient.ListPods()` fails, but our cli successfully made the grpc call to conduit-api, wouldn't we want to return grpc success, with a failed `telemetryClientCheck` in the response? or am i reading this wrong?